### ISSUE TITLE
[Swift Codegen] Protocol 'SelectionSet' cannot be nested inside another declaration

### ISSUE THREAD

**Issue by kimdv** [apollographql/apollo-ios#2293]
## Bug report

I'm using the latest alpha of the iOS SDK, and the Swift code generation. 

The generated result look like this: 

```swift
public extension GraphQLApi {
  public typealias ID = String

  public protocol SelectionSet: ApolloAPI.SelectionSet & ApolloAPI.RootSelectionSet
  where Schema == GraphQLApi.Schema {}

  public protocol InlineFragment: ApolloAPI.SelectionSet & ApolloAPI.InlineFragment
  where Schema == GraphQLApi.Schema {}

  public enum Schema: SchemaConfiguration {
    public static func objectType(forTypename __typename: String) -> Object.Type? {
      switch __typename {
      case "Query": return GraphQLApi.Query.self
      default: return nil
      }
    }
  }
}
```

this gives me a compilation error: `Protocol 'SelectionSet' cannot be nested inside another declaration`

## Versions

Please fill in the versions you're currently using: 

- `apollo-ios` SDK version: Alpha-5
- Xcode version: 13.4
- Swift version: 5.6.1
- Package manager: Yes

## Steps to reproduce

Our generation command

```swift
  struct GenerateCode: ParsableCommand {
    static var configuration = CommandConfiguration(
      commandName: "generate",
      abstract: "Generates swift code from your schema + your operations based on information set up in the `GenerateCode` command.")

    func run() throws {
      /// The root of the target for which you want to generate code.
      let targetRootURL = SourceRootURL
        .apollo.childFolderURL(folderName: "customer-ios/graphql")

      /// The URL where the generated schema files will be written to.
      let schemaModuleURL = SourceRootURL
        .apollo.childFolderURL(folderName: "customer-ios/graphql")

      // Make sure the folders exists before trying to generate code.
      try FileManager.default.apollo.createDirectoryIfNeeded(atPath: targetRootURL.path)
      try FileManager.default.apollo.createDirectoryIfNeeded(atPath: schemaModuleURL.path)

      // Create the Codegen configuration object. For all configuration parameters see: https://www.apollographql.com/docs/ios/api/ApolloCodegenLib/structs/ApolloCodegenConfiguration/
      let codegenConfiguration = ApolloCodegenConfiguration(
        schemaName: "GraphQLApi", input: ApolloCodegenConfiguration.FileInput(schemaPath: SchemaOutputURL.path, searchPaths: ["\(targetRootURL.path)/**/*.graphql"]),
        output: ApolloCodegenConfiguration.FileOutput(
          schemaTypes: ApolloCodegenConfiguration.SchemaTypesFileOutput(
            path: schemaModuleURL.path,
            moduleType: .embeddedInTarget(name: "customer-ios"))
        )
      )

      // Actually attempt to generate code.
      try ApolloCodegen.build(with: codegenConfiguration)
    }
  }
```

**Comment by kimdv** [comment]
@AnthonyMDev If you need a test/feedback before making a release feel free to reach out here 🙌

**Comment by calvincestari** [comment]
@kimdv - the fix for this is merged and will be available in the next alpha release, 1.0.0-alpha.6. Keep an eye out for that later today.
